lib/pos_consensus_controller.go

```diff
@@ -55,9 +56,77 @@ func (cc *ConsensusController) HandleFastHostStuffBlockProposal(event *consensus
 	cc.lock.Lock()
 	defer cc.lock.Unlock()
 
-	if !consensus.IsProperlyFormedConstructVoteQCEvent(event) {
-		// If the event is not properly formed, we ignore it and log it. This should never happen.
-		return errors.Errorf("HandleFastHostStuffBlockProposal: Received improperly formed vote QC construction event: %v", event)
+	// Handle the event as a block proposal event for a regular block
+	err := cc.handleBlockProposerEvent(event, consensus.FastHotStuffEventTypeConstructVoteQC)
+	if err != nil {
+		return errors.Wrapf(err, "HandleFastHostStuffBlockProposal:")
+	}
+
+	// Happy path: nothing left to do
+	return nil
+}
+
+// HandleFastHostStuffBlockProposal is called when FastHotStuffEventLoop has signaled that it can
+// construct a block with a timeout QC at a certain block height. This function validates the block
+// proposal signal, constructs, processes locally, and then and broadcasts the block.
+//
+// Steps:
+//  1. Validate that the block height and view we want to propose the block with are not stale
```

Comment:
It's hard to see these steps in this function. Should this section of the comment be moved to `handleBlockProposerEvent`?

Reply:
Sounds good. I moved the steps to the helper function instead